src/Spaces/dss.jl

```diff
@@ -657,62 +657,291 @@ function dss_2d!(
     return data
 end
 
-function weighted_dss!(
+"""
+    weighted_dss!(data, space, ghost_buffer = nothing)
+
+Computes weighted dss of `data`. This function consists of 
+
+1). `weighted_dss_start!` which loads the send buffer and starts communications,
+
+2). `weighted_dss_internal!` which progresses the communication and performs dss operations on interior vertices and faces, and
+
+3). `weighted_dss_ghost!` which completes the communication and performs dss on ghost vertices and faces
+
+These constituent functions can also be called separately for increased overlap between computation and communication when merging multiple dss
+operations.
+
+"""
+function weighted_dss!(data, space, ghost_buffer = nothing)
+    weighted_dss_start!(data, space, ghost_buffer)
+    weighted_dss_internal!(data, space, ghost_buffer)
+    weighted_dss_ghost!(data, space, ghost_buffer)
+end
+
+"""
+    weighted_dss_start!(data, space, ghost_buffer = nothing)
+
+Create the ghost buffer, if necessary, load the send buffer and start communication.
+
+Part of [`Spaces.weighted_dss!`](@ref).
+"""
+function weighted_dss_start!(
+    data,
+    space::ExtrudedFiniteDifferenceSpace{S, H},
+    ghost_buffer = nothing,
```

Comment:
Don't provide a default argument (and update the docs)
```suggestion
    ghost_buffer,
```